packages/node/src/behavior/state/managed/ManagedReference.ts

```diff
@@ -51,10 +53,24 @@ export function ManagedReference(
     const key = primaryKey === "id" ? (id ?? name) : name;
     const altKey = primaryKey === "id" ? (key === name ? undefined : name) : id;
     let value: unknown;
-    if (key in (parent.value as Container)) {
-        value = (parent.value as Container)[key];
-    } else if (altKey !== undefined) {
-        value = (parent.value as Container)[altKey];
+
+    let properties: Val.Struct | undefined;
```

Comment:
Maybe call this `dynamicContainer` or something, then it'll be a little clearer what the code's doing.